ui/packages/ui/src/main/webapp/js/ServiceFactoryNameRegistry.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define*/
+define([
+], function () {
+
+    return {
+        getName: function(properties) {
+            switch(properties.get('service.factoryPid')) {
+                case 'org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor':
+                    return properties.get('monitoredDirectoryPath');
```

Comment:
Since we have custom logic for the directory monitor here now I wonder if we should include the processingMechanism in this name as well. `in_place  /dir/path/monitored`

Reply:
If it's something that would help admins we should.